R/models.R

```diff
@@ -0,0 +1,261 @@
+                      "gddUnits", "id", "stage", "description", "gddThreshold")
+  rownames(data) <- c(1:nrow(data))
+  
+  data <- dplyr::mutate_if(data, is.factor, as.character)
+  
+  
+  if(nrow(data) == 0) {
+    stop(a$simpleMessage)
+  } else {
+    return(as.data.frame(data))
+  }
+}
+
+
+
+
+
+
+#' @title Get Model Results
+#'
+#' @description
+#' \code{get_model_details} Calls Model Results endpoint of API
+#'
+#' @details
+#' Running a model is as simple as requesting its current results. The aWhere platform
+#' will use the information previously provided about a field and planting to generate
+#' the relevant, field-specific information. For growth stage models, the results are
+#' the current stage as well as information about the most recent and next stage.
+#' 
+#' Important: Models require the use of the Fields and Plantings APIs to store important
+#' information about a field and crop. When a Field is specified for this API, it will
+#' select the most recently created Planting record to retrieve the crop and planting date.
+#'
+#' @references https://developer.awhere.com/api/reference/models/results
+#'
+#' @param - model_id: A model id to retrieve details for that specific model
+#' @param - keyToUse: aWhere API key to use.  For advanced use only.  Most users will not need to use this parameter (optional)
+#' @param - secretToUse: aWhere API secret to use.  For advanced use only.  Most users will not need to use this parameter (optional)
+#' @param - tokenToUse: aWhere API token to use.  For advanced use only.  Most users will not need to use this parameter (optional)
+#'
+#' @import httr
+#'
+#' @return - data.frame containing details about requested model
+#'
+#' @examples
+#' \dontrun{get_model_results('Field1', 'BarleyGenericMSU')
+#' }
+
+#' @export
+
+get_model_results <- function(field_id
+                              ,model_id
+                              ,keyToUse = awhereEnv75247$uid
+                              ,secretToUse = awhereEnv75247$secret
+                              ,tokenToUse = awhereEnv75247$token) {
+  
+  checkCredentials(keyToUse,secretToUse,tokenToUse)
+  
+  ## Create Request
+  url <- "https://api.awhere.com/v2/agronomics/fields/"
+  
+  url <- paste0(url, field_id, "/models/", model_id, "/results")
+  
+
+  doWeatherGet <- TRUE
+  while (doWeatherGet == TRUE) {
+    request <- httr::GET(url,
+                         httr::content_type('application/json'),
+                         httr::add_headers(Authorization =
+                                             paste0("Bearer ", tokenToUse)))
+    
+    a <- suppressMessages(httr::content(request))
+    
+    doWeatherGet <- aWhereAPI:::check_JSON(a,request)[[1]]
+  }
+  
+  
+  a[sapply(a, is.null)] <- NA
+  
+  data <- data.frame(biofixDate = a$biofixDate, nextStage = as.character(a$nextStage), gddUnits = a$gddUnits, modelId = a$modelId, latitude = a$location$latitude,
```

Comment:
when I do the above commands this is the contents of the named list

names(a)
[1] "biofixDate"     "previousStages" "currentStage"   "nextStage"      "gddUnits"       "modelId"        "location"       "plantingDate"  
[9] "_links"